src/providers/okta.provider.ts

```diff
@@ -267,17 +283,33 @@ export class OktaProvider {
     }
 
     static async deleteUser(ctx: Context): Promise<void> {
+        logger.info(`[OktaProvider] - Delete user with id ${ctx.params.id}`);
         try {
-            logger.info(`[OktaProvider] - Delete user with id ${ctx.params.id}`);
-            const deletedUser: IUser = await OktaService.deleteUser(ctx.params.id);
-            ctx.body = UserSerializer.serialize(deletedUser);
+            await OktaService.getOktaUserById(ctx.params.id);
         } catch (err) {
             if (err instanceof UserNotFoundError) {
-                return ctx.throw(404, 'User not found');
+                ctx.throw(404, 'User not found');
+                return;
             }
 
-            logger.error('[OktaProvider] - Error updating my user, ', err);
+            logger.error('[OktaProvider] - Error loading user for deletion, ', err);
+            ctx.throw(500, 'Internal server error');
+        }
+
+        let deletedUser: IUser = null;
+        try {
+            deletedUser = await OktaService.deleteUser(ctx.params.id);
+            ctx.body = UserSerializer.serialize(deletedUser);
+        } catch (err) {
+            logger.error('[OktaProvider] - Error deleting user, ', err);
             return ctx.throw(500, 'Internal server error');
+        } finally {
+            const newDeletionData: Partial<IDeletion> = {
+                userId: ctx.params.id,
+                requestorUserId: Utils.getUser(ctx).id,
+                userAccountDeleted: deletedUser !== null,
+            };
+            await DeletionService.createDeletion(newDeletionData);
```

Comment:
I saw this mentioned in [your JIRA comments](https://gfw.atlassian.net/browse/GTC-1881?focusedCommentId=13242). Good call. 💪